File: 125-valid-palindrome/solution.py
# Runtime: 33ms, Beats 98.51% of users with Python3
# Memory: 17.79MB, Beats 60.73% of users with Python3



class Solution:
    def isPalindrome(self, s: str) -> bool:
        l=re.findall("[a-zA-Z0-9]",s)
        p=0
        q=len(l)-1
        while(p<q):
            if(l[p].lower()!=l[q].lower()):
                return False
            p+=1
            q-=1
        return p>=q
        

File: 125-valid-palindrome/solution.cpp
// Runtime: 0ms, Beats 100.00% of users with C++
// Memory: 8.60MB, Beats 13.78% of users with C++

class Solution {
public:
    bool isPalindrome(string s) {
        int p=0;
        int q=s.length()-1;
        while(p<q)
        {
            while(!isalnum(s[p]) && p<q) p++;
            while(!isalnum(s[q]) && p<q) q--;
            if(tolower(s[p])!=tolower(s[q])) return false;
            p++;
            q--;
        }
        return true;
    }
};